R/helpfunctions.R

```diff
@@ -1,70 +1,64 @@
-}
-
-
-is_stan_model <- function(fit) {
-  inherits(fit, c("stanreg", "stanfit", "brmsfit"))
-}
-
-
-#' @importFrom sjmisc str_contains
-get_glm_family <- function(fit) {
-  c.f <- class(fit)
-
-  # do we have glm? if so, get link family. make exceptions
-  # for specific models that don't have family function
-  if (any(c.f %in% c("lme", "plm"))) {
-    fitfam <- ""
-    logit_link <- FALSE
-  } else {
-    fitfam <- stats::family(fit)$family
-    logit_link <- stats::family(fit)$link == "logit"
-  }
-
-  # create logical for family
-  binom_fam <- fitfam %in% c("binomial", "quasibinomial")
-  poisson_fam <- fitfam %in% c("poisson", "quasipoisson") ||
-    sjmisc::str_contains(fitfam, "negative binomial", ignore.case = T)
-
-  list(is_bin = binom_fam, is_pois = poisson_fam, is_logit = logit_link)
-}
-
-# return names of objects passed as ellipses argument
-dot_names <- function(dots) unname(unlist(lapply(dots, as.character)))
+# Help-functions
+
+data_frame <- function(...) {
+  x <- data.frame(..., stringsAsFactors = FALSE)
+  rownames(x) <- NULL
+  x
+}
+
+
+is_merMod <- function(fit) {
+  inherits(fit, c("lmerMod", "glmerMod", "nlmerMod", "merModLmerTest"))
+}
+
+
+is_stan_model <- function(fit) {
+  inherits(fit, c("stanreg", "stanfit", "brmsfit"))
+}
+
+
+#' @importFrom sjmisc str_contains
+get_glm_family <- function(fit) {
+  c.f <- class(fit)
+
+  # do we have glm? if so, get link family. make exceptions
+  # for specific models that don't have family function
+  if (any(c.f %in% c("lme", "plm"))) {
+    fitfam <- ""
+    logit_link <- FALSE
+  } else {
+    fitfam <- stats::family(fit)$family
+    logit_link <- stats::family(fit)$link == "logit"
+  }
+
+  # create logical for family
+  binom_fam <- fitfam %in% c("binomial", "quasibinomial")
+  poisson_fam <- fitfam %in% c("poisson", "quasipoisson") ||
+    sjmisc::str_contains(fitfam, "negative binomial", ignore.case = T)
+
+  list(is_bin = binom_fam, is_pois = poisson_fam, is_logit = logit_link)
+}
+
+# return names of objects passed as ellipses argument
+dot_names <- function(dots) unname(unlist(lapply(dots, as.character)))
 
 
 #' @importFrom tidyr nest
 #' @importFrom dplyr select filter group_vars
-#' @importFrom stats complete.cases
-#' @importFrom rlang .data
-get_grouped_data <- function(x) {
-  # nest data frame
-  grps <- tidyr::nest(x)
-
-  # remove NA category
-  cc <- grps %>%
-    dplyr::select(-.data$data) %>%
-    stats::complete.cases()
-
-  # select only complete cases
-  grps <- grps %>% dplyr::filter(!! cc)
-
-  # arrange data
-  if (length(dplyr::group_vars(x)) == 1)
-    reihe <- order(grps[[1]])
-  else
-    reihe <- order(grps[[1]], grps[[2]])
-  grps <- grps[reihe, ]
-
-  grps
-}
+#' @importFrom stats complete.cases
+#' @importFrom rlang .data
+get_grouped_data <- function(x) {
+  # retain observations that are complete wrt grouping vars, then nest
+  grps <- x %>%
+    dplyr::group_modify(~ filter(.x, complete.cases(.y))) %>%
```

Comment:
Thanks! Do `filter()` and `complete.cases()` require namespace reference (`::`) inside the anonymous function (`dplyr::filter()` `and stats::complete.cases()`)?

Reply:
I noticed that you have already imported `dplyr::filter()` into the NAMESPACE, hence my non-qualified call:

https://github.com/strengejacke/sjstats/blob/5a647a515d9d44d2ad4882199591eea78370ec04/R/helpfunctions.R#L47

Generally, it's either/or: don't import the functions in your namespace and call as `pkg::foo()` or do import them, then call as `foo()`.

You've listed `stats` in Depends, so I did not namespace `complete.cases()` either. If it were in Imports, I would have.

https://github.com/strengejacke/sjstats/blob/master/DESCRIPTION#L27